staging/src/k8s.io/apiserver/pkg/endpoints/handlers/response.go

```diff
@@ -510,15 +510,20 @@ type watchListTransformer struct {
 	targetGVK                  *schema.GroupVersionKind
 	negotiatedEncoder          runtime.Encoder
 	buffer                     runtime.Splice
+
+	// kcp: needed for setKCPOriginalAPIVersionAnnotation().
+	// It expects a context with clusterContextKey key set.
+	ctx context.Context
```

Comment:
:'( one more ctx wiring. Maybe we could try to get this upstreamed too as its apiserver code.